modules/shmnet.py

```diff
@@ -32,13 +32,14 @@ def forward(self, batch):
             gold_trimaps = batch['trimap'].to(self.args.device)
 
             # Convert to BUF (BG, UNC, FG) classes.
-            gold_trimaps = gold_trimaps.repeat(1, 3, 1, 1)
+            gold_trimaps = gold_trimaps.repeat(1, 3, 1, 1) * 255
             gold_trimaps[:, 0, :, :] = gold_trimaps[:, 0, :, :].eq(0).float()
             gold_trimaps[:, 1, :, :] = gold_trimaps[:, 1, :, :].eq(128).float()
             gold_trimaps[:, 2, :, :] = gold_trimaps[:, 2, :, :].eq(255).float()
 
             if self.args.mode != 'end_to_end':
                 # Teacher forcing
+                _, unsure, foreground = torch.split(gold_trimaps, 1, dim=1)
```

Comment:
`pred_mattes` will be valid only when `unsure` and `foreground` are continuous values.
https://github.com/nikhilweee/semantic-human-matting/blob/a0f1b1ff65cb8fbd1d78e25fc7da7d223ef97cf7/modules/shmnet.py#L49
This change makes them discrete, which I believe might be a problem.

Reply:
If the output of T-Net is used for fusion, the weight of T-Net will be updated when M-Net is trained separately. 
I think there should be no problem. Just like trimap-based matting algorithm, it only uses the result of unsure region, the known region is covered by mask.
Same as:
```python
pred_mattes[trimap == 0] = 0
pred_mattes[trimap == 255] = 1
```

Reply:
Fair enough. I'll take that. Thanks again!